packages/datadog_tracking_http_client/lib/src/tracking_http_client.dart

```diff
@@ -107,17 +107,20 @@ class DatadogTrackingHttpClient implements HttpClient {
     final rum = datadogSdk.rum;
 
     if (rum != null) {
-      try {
-        rumKey = uuid.v1();
-        final rumHttpMethod = rumMethodFromMethodString(method);
-        rum.startResource(rumKey, rumHttpMethod, url.toString());
-      } catch (e, st) {
-        datadogSdk.internalLogger.sendToDatadog(
-          '$DatadogTrackingHttpClient encountered an error while attempting '
-          ' to track an _openUrl call: $e',
-          st,
-          e.runtimeType.toString(),
-        );
+      bool shouldTrack = _shouldTrackUrl(url);
+      if (shouldTrack) {
+        try {
+          rumKey = uuid.v1();
+          final rumHttpMethod = rumMethodFromMethodString(method);
+          rum.startResource(rumKey, rumHttpMethod, url.toString());
+        } catch (e, st) {
+          datadogSdk.internalLogger.sendToDatadog(
+            '$DatadogTrackingHttpClient encountered an error while attempting '
```

Comment:
otherwise there is double space between `attempting` and `to`

```suggestion
            '$DatadogTrackingHttpClient encountered an error while attempting'
```